llvm/unittests/Target/AArch64/SMEAttributesTest.cpp

```diff
@@ -82,19 +102,86 @@ TEST(SMEAttributes, Basics) {
   // Test PSTATE.ZA interfaces.
   ASSERT_FALSE(SA(SA::ZA_Shared).hasPrivateZAInterface());
   ASSERT_TRUE(SA(SA::ZA_Shared).hasSharedZAInterface());
+  ASSERT_TRUE(SA(SA::ZA_Shared).sharesZA());
   ASSERT_TRUE(SA(SA::ZA_Shared).hasZAState());
   ASSERT_FALSE(SA(SA::ZA_Shared).preservesZA());
   ASSERT_TRUE(SA(SA::ZA_Shared | SA::ZA_Preserved).preservesZA());
 
   ASSERT_TRUE(SA(SA::ZA_New).hasPrivateZAInterface());
+  ASSERT_FALSE(SA(SA::ZA_New).hasSharedZAInterface());
```

Comment:
nit: add this case too?
```
ASSERT_FALSE(SA(SA::ZA_Shared).sharesZA());
```

Reply:
There is already a `sharesZA()` test for `SA::ZA_Shared` on line 105, did you mean:
`ASSERT_FALSE(SA(SA::ZA_Shared).sharesZT0())`?

I've added `sharesZT0()` cases for `SA::ZA_Shared` & `SA::ZA_New` in the latest commit.

Reply:
I'm not sure why I asked for this yesterday, perhaps my eyes just didn't spot this case.